website/webresources/js/app/shopping-cart-details/shopping-cart-details.jsx

```
@@ -70,19 +69,57 @@ class ShoppingCartDetails extends Component {
                 }
             }
         }
-        window.Abc.order.removeCartItems(this.state.cartMeta, lineItemIdArray).then(res1 => {
-            if (!res1.error) {
+        window.Abc.order.removeCartItems(this.state.cartMeta, lineItemIdArray).then(res => {
+            if (!res.error) {
+                let cartMetaOjb = { id:res.id, version:res.version };
                 this.setState({
-                    fullCart: res1,
-                    cartMeta: { id:res1.id, version:res1.version }
+                    fullCart: res,
+                    cartMeta: cartMetaOjb
                 });
+                if(res.lineItems.length == 0 && res.discountCodes.length > 0) {
+                    // Remove discount code stuff here...
+                    let discountCodeId = res.discountCodes[0].discountCode.id;
+                    this.removeDiscountCoupon(discountCodeId, cartMetaOjb);
+                }
             }
-            if (res1.statusCode == 404) { // res.statusCode == 404 if no cart exists
-                this.setState({ cartMeta: { id:res1.id, version:res1.version } });
+            if (res.statusCode == 404) { // res.statusCode == 404 if no cart exists
+                this.setState({ cartMeta: { id:res.id, version:res.version } });
             }
         });
 
     }
+
+    /**
+     * Removal of discount-code
+     * @param {String} discountCodeId 
+     */
+    removeDiscountCoupon(discountCodeId, cartMetaOjb) {
+        let cartMeta = cartMetaOjb;
+        let actions = [
+            {
+                'action':'removeDiscountCode',
+                'discountCode':{
+                    'typeId': 'discount-code',
+                    'id': discountCodeId || ''
+                }
+            }
+        ];
+        
+        return removeDiscountCodeOnCart(cartMeta, actions).then(res => {
+            if(res.id) {
```

@Jayprakash-manjhi1987  for future perspective please add a else block in which we can handle error through error handler.

Your approach here not to show any error to user is correct since this should be silent failure